### PR TITLE
[flutter_tools] Remove web specific vm_service handlers, move handler tests to single location

### PR DESCRIPTION
**Pull request by jonahwilliams**
The goal of this PR is to move the resident runners closer to a state where enough of the implementation is shared to enable 1) profile/release/debug autodetect for `flutter attach` and 2) simultaneous connections to web and non-web devices. I also wanted to remove the vast amounts of duplicated test logic due to the web and non-web resident runners having different service extension impls as well as remove the mocks from the terminal handler test and re-imagine it as a fairly high level interaction test.

Changes in resident_runner.dart:

* Creates a new class `ResidentHandlers` which contains the vm service delegation, make it easier to share with the web impl for now. This required exposing Logger and FileSystem as protected members.
* Removes all of the wrapper vm service functionality from the `FlutterDevice` class. Now that the logic is in a single place it doesn't make as much sense to split it up.
* Updates FlutterDevice to conditionally use `getVM`'s isolate list instead of listing the flutter views. The flutter view RPC is not supported on the web, but since there is only ever one isolate we can use the VM list. This required exposing the TargetPlatform field on FlutterDevice, which was already part of the constructor.

Changes in resident runner testing:

* Deleted all tests that only checked that service extensions were called correctly

Changes in resident_web_runner testing:

* Deleted all tests that only checked that service extensions were called correctly

Changes in terminal_handler testing:

Updated the test set up to remove the use of mocks, and to extend most test cases to include a web case.

clean ups:

* Remove the requirement to inject fsUtils to grab the unique file functionality, since all this needs is a FileSystem.
* Removes the feature flag check for debug target platform. I think its OK to let people toggle to macOS .

### CODE REVIEW

File: packages/flutter_tools/lib/src/isolated/resident_web_runner.dart [hunk 1]
```diff
@@ -230,242 +214,12 @@ class ResidentWebRunner extends ResidentRunner {
     _logger.printStatus('For a more detailed help message, press "h". $quitMessage');
   }
 
-  @override
-  Future<bool> debugDumpApp() async {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/isolated/resident_web_runner.dart:234]: All of these now use the same implementation as the "regular" resident runner.

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 2]
```diff
@@ -400,138 +401,16 @@ class FlutterDevice {
     return devFS.create();
   }
 
-  Future<void> debugDumpApp() async {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/resident_runner.dart:403]: Inlined all of these into the resident runner.

File: packages/flutter_tools/lib/src/resident_runner.dart [hunk 3]
```diff
@@ -849,38 +1130,14 @@ abstract class ResidentRunner {
...
+  bool get supportsRestart {
+    return isRunningDebug && flutterDevices.every((FlutterDevice device) {
       return device.device.supportsHotRestart;
     });
   }
 
-  /// Invoke an RPC extension method on the first attached ui isolate of the first device.
-  // TODO(jonahwilliams): Update/Remove this method when refactoring the resident
-  // runner to support a single flutter device.
-  Future<Map<String, dynamic>> invokeFlutterExtensionRpcRawOnFirstIsolate(
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/resident_runner.dart:866]: inlined into the remaining use site in daemon.dart

**Comment by jonahwilliams** [comment]
Rebased on the previous PR, now this is only the test re-arrangement and the removal of the web specific vm service delegation

**Review by jmagman — APPROVED** [review]
Thanks, this was much easier to review in two pieces.
LGTM!

**Pull request merged**